### DISCUSSION TITLE
How to use resolve: {alias} config

### DISCUSSION THREAD

**Discussion by RailsCod3rFuture** [shakacode/shakapacker#452] [Q&A] [answered]
I'm getting an odd isssue where shakapacker is looking for my aliases in the node_modules folder instead of src folder in my javascript folder. I removed all configurations that might relate to this issue...and no cigar. How do I alias files using webpack.config.js?

**Comment by RailsCod3rFuture** [comment]
alias doesn't work with @ symbols -- it seems.

**Reply by ahangarha** [reply]
Can you provide a public demo project so that we can look into the issue?

**Comment by RailsCod3rFuture** [comment]
The most I can show you is my setup -- given that the code related to it is not my personal project. But I recently upgraded to shakapacker from webpacker for it.

in js file

depedencyJs.js

`import {MY_ACTION} from '@actions/types'`

webpack.config.js
```

const path = require('path')
const {generateWebpackConfig} = require('shakapacker')

const webpackConfig = generateWebpackConfig({
entry: ['@babel/polyfill'],
resolve: {
alias: {
  '@actions': path.resolve(__dirname, '../../app/javascript/src/actions')
},
extensions: [".js", ".ts", ".jsx", ".tsx"]
},
})

module.exports = webpackConfig
```

**Reply by tomdracz** [reply]
Tried it here https://github.com/tomdracz/shakapacker-css-test/commit/97598b2e8f6bc011934a67b52ebcda007dd00f28 but unable to reproduce. Alias resolves correctly and pulls in with @ symbol.

If you are able to reproduce using the toy repo as linked above, we can have a further look but sounds like something else in your setup is causing this potentially @RailsCod3rFuture

**Reply by RailsCod3rFuture** [reply]
Can you upload the repo and show me your setup? I can probably check to see if some other config in the app is causing this to break for me

**Reply by tomdracz** [reply, marked as answer]
Whole repo available at https://github.com/tomdracz/shakapacker-css-test/tree/alias-test

**Reply by RailsCod3rFuture** [reply]
I'm beginning to believe that its either an M1 issue or user rights issue...

**Reply by tomdracz** [reply]
For what it's worth, I'm running M2 mac so I would discard that.

Might be worth console logging in that config file the `path.resolve(__dirname, '../../app/javascript/src/actions')` and `__dirname` itself to see if it points at some wrong resolution.

Other thoughts:
- have you tried any other @ aliases? Is there some directory starting with @actions in node modules that could be tripping the resolution?
- I remember a while back having to use additional plugin for Babel to respect aliases but not sure if that's still relevant 🤔
